src/main/scala/com/sumologic/sumobot/core/Receptionist.scala

```diff
@@ -110,49 +108,53 @@ class Receptionist(rtmClient: SlackRtmClient,
       pendingIMSessionsByUserId = pendingIMSessionsByUserId + (userId -> (doneRecipient, doneMessage))
 
     case message: Message if !tooOld(message.ts, message) =>
-      translateAndDispatch(message.channel, message.user, message.text, message.ts)
+      translateAndDispatch(message.channel, message.user, message.text, message.ts, parentTimeStamp = message.thread_ts)
 
     case messageChanged: MessageChanged if !tooOld(messageChanged.ts, messageChanged) =>
       val message = messageChanged.message
       translateAndDispatch(messageChanged.channel, message.user, message.text, message.ts)
 
     case botMessage: BotMessage if !tooOld(botMessage.ts, botMessage) && botMessage.username.isDefined =>
-        translateAndDispatch(botMessage.channel, botMessage.username.get, botMessage.text, botMessage.ts,
-                             fromBot = true,
-                             botMessage.attachments)
+      translateAndDispatch(botMessage.channel, botMessage.username.get, botMessage.text, botMessage.ts,
+        fromBot = true,
+        botMessage.attachments)
 
     case RtmStateRequest(sendTo) =>
       sendTo ! RtmStateResponse(rtmClient.state)
   }
 
   protected def translateMessage(channelId: String, idTimestamp: String, from: Sender, text: String,
-                                 attachments: Seq[IncomingMessageAttachment]): IncomingMessage = {
+                                 attachments: Seq[IncomingMessageAttachment],
+                                 parentTimeStamp: Option[String] = None): IncomingMessage = {
```

Comment:
I think we should add it after `idTimestamp`, not at the end (here and in `IncomingMessage`).